yaksh/models.py

```diff
@@ -1077,11 +1083,16 @@ def regrade(self, question_id):
         json_data = question.consolidate_answer_data(answer) \
                             if question.type == 'code' else None
         correct, result = self.validate_answer(answer, question, json_data)
-        user_answer.marks = question.points if correct else 0.0
         user_answer.correct = correct
         user_answer.error = result.get('error')
+        if correct:
```

Comment:
If condition will have to stay.
In case when there is no partial grading for a question, in that case the evaluator always returns 0.0
So if a question is answered correctly but it is not graded partially then it will end up being marked 0.0 marks (in case success=True is not checked)